### PR TITLE
Upgrading upstream dep to avoid vulnerability

### LINKED ISSUE #27: json-schema dependency
The dependency json-schema has licensing issues (see https://github.com/kriszyp/json-schema/issues/62 and https://github.com/kriszyp/json-schema/issues/69), therefore it is also unclear whether it is allowed to use that code in combination with the MIT license of jsprim.

As that package seems to be unmaintained as well (# of pull requests and open issues), please consider whether this dependency can be dropped, or a more active fork could be used instead.

### PR DESCRIPTION
**Pull request by JJ**
Prototype polution: https://security.snyk.io/vuln/SNYK-JS-JSONSCHEMA-1920922
Incorporates #27 since it's got a better representation for the remaining deps.

> Thus closes #27 when merged.

### CODE REVIEW

**Review by BruceHaley — APPROVED** [review]
This is great! Our Bot Framework samples have a nested dependency on prim. Looking forward to upgrading to the prim version that carries this json-schema v 0.4.0 dependency, thereby fixing the vulnerability. Any ETA on when the patch will be available?

**Review by jschirrmacher — APPROVED** [review]

**Comment by JJ** [comment]
@bahamat any feedback on this?

File: package.json [hunk 1]
```diff
@@ -8,10 +8,10 @@
...
 	},
 	"dependencies": {
-		"assert-plus": "1.0.0",
-		"extsprintf": "1.3.0",
-		"json-schema": "0.2.3",
-		"verror": "1.10.0"
+		"assert-plus": "^1.0.0",
+		"extsprintf": "^1.3.0",
+		"json-schema": "^0.4.0",
+		"verror": "^1.10.0"
```
Comment (Morl99) [inline package.json:14]: Why are you adding carets? If you think this is a good idea, you should separate this into a different PR, otherwise you risk that this PR will get lost in discussion due to that unexpected change. Or was that merely by mistake?

**Comment by bahamat** [comment]
Fixed in 017f744

File: package.json (hunk 1 above)
Comment (JJ) [reply package.json:14]: Got it from the previous PR.
